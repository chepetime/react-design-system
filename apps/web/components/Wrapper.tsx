import React from "react";

interface iProps {
  children: React.ReactNode;
}

export default function Wrapper({ children }: iProps) {
  return (
    <div className="Wrapper" style={{ padding: "0 2rem", margin: "0 auto" }}>
      {children}
    </div>
  );
}
