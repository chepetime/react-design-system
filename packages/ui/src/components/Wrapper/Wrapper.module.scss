@use "./../../styles" as styles;

.Wrapper {
  @include styles.styles-reset;
  @include styles.text-wrapping;
  width: 100%;

  max-width: 75rem;
  margin: 0 auto;
  padding: 0 1rem;

  @include styles.respond-to("md") {
    padding: 0 2rem;
  }
}
