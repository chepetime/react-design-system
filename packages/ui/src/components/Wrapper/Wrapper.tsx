import * as React from "react";
import classnames from "classnames";
import css from "./Wrapper.module.scss";

export interface WrapperProps {
  children?: React.ReactNode;
}

export const Wrapper = React.forwardRef<HTMLDivElement, WrapperProps>(
  (props: WrapperProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.Wrapper)} ref={ref}>
        {props.children}
      </div>
    );
  }
);

Wrapper.displayName = "Wrapper";
