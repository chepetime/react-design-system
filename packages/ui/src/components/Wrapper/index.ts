export { Wrapper } from "./Wrapper";
export type { WrapperProps } from "./Wrapper";