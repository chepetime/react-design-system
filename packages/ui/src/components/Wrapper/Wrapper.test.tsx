import React from "react";import { render } from "@testing-library/react";
import { Wrapper } from "./Wrapper";

describe("Wrapper", () => {
  test("renders the Wrapper component", () => {
    render(<Wrapper />);
  });
});
