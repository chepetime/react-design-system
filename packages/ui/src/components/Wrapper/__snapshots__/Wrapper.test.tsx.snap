// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper matches snapshot 1`] = `
<div>
  <div
    class="Wrapper"
  />
</div>
`;
